member.php

```diff
@@ -1599,8 +1600,29 @@
 			$password_length = 8;
 		}
 
-		$password = random_str($password_length);
-		$logindetails = update_password($user['uid'], md5($password), $user['salt']);
+		// Set up user handler.
+		require_once 'inc/datahandlers/user.php';
+		$userhandler = new UserDataHandler('update');
+
+		$userhandler->set_data(array(
+			'uid'		=> $user['uid'],
+			'username'	=> $user['username'],
+			'email'		=> $user['email'],
+			'password'	=> random_str($password_length)
+		));
+
+		if(!$userhandler->validate_user())
+		{
+			error($lang->error_unknownerror);

```

Comment:
Why "Unknown error."? This won't help the user with anything. `get_friendly_errors()` could be used. Imploded with `<br />` if you don't want to switch to `inline_error()`.